### PR TITLE
Componentize the hamburger menu.

### PR DESCRIPTION
**Pull request by jackconwayiv**
Pulls out navigation via hamburger menu and keystroke bindings into a separate component from NavBar.

### CODE REVIEW

File: frontend/public/manifest.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "short_name": "React App",
-  "name": "Create React App Sample",
+  "short_name": "livepowerhour.com",
+  "name": "songbook generator",
```
Comment (samdatkins) [inline frontend/public/manifest.json:3]: Doesn't matter for now, but before we suggest people install the app we should review the [icon guidelines](https://web.dev/learn/pwa/web-app-manifest/) and probably title case the name

File: frontend/src/components/NavBar.tsx [hunk 2]
```diff
@@ -188,89 +107,16 @@ export default function NavBar({
...
+            timerControls={timerControls}
+            isLive={isLive}
+            setIsLive={setIsLive}
+            addSongDrawerOutlet={addSongDrawerOutlet}
+            asyncSongbook={asyncSongbook}
+            resetAppState={resetAppState}
+          />
         </Flex>
 
         {!isMobileDevice && (
```
Comment (samdatkins) [inline frontend/src/components/NavBar.tsx:122]: Feels much better to cut down the size of this file!

**Review by samdatkins — APPROVED** [review]

File: frontend/public/manifest.json (hunk 1 above)
Comment (jackconwayiv) [reply frontend/public/manifest.json:3]: Noted!

File: frontend/src/components/NavBar.tsx (hunk 2 above)
Comment (jackconwayiv) [reply frontend/src/components/NavBar.tsx:122]: I'm wondering about the viability of pulling the nav part(s) of the hamburger menu out into a separate component. That would enable us to construct it as needed for each various page we're using it on.

**Pull request merged**